### ISSUE TITLE
Watch Mode Plugin: jest-runner-eslint/watch

### ISSUE THREAD

**Issue by azz** [jest-community/jest-runner-eslint#43]
Combining https://github.com/facebook/jest/issues/5212 and the new Jest Watch Plugins API, it would be really cool to have:

```json
{
  "jest": {
    "runner": "jest-runner-eslint",
    "watchPlugins": [
      "jest-runner-eslint/watch"
    ]
  }
}
```

Which, when an ESLint error is reported, exposes an option in watch mode:

```
Press F to run eslint --fix
```

I think it makes sense for it to live in this repo, but it could equally be a standalone repo.

I'd be happy to work on a PR if it is desired here 😄 

Related: #10

**Comment by ljharb** [comment]
I think it makes sense here.

**Comment by rogeliog** [comment]
Yes, this sounds really interesting! Currently, I don't think that there is a good way of sending that information from the plugin to the runner. It would be awesome find a way that could enable this type of features!

**Comment by rogeliog** [comment]
I think I have some vague idea, I'll try to hack something today

**Comment by rogeliog** [comment]
I think having it in the same repo would make a lot of sense, given that it is a watch plugin really specific to this runner.


#### Here is a PoC:
There are still a couple of hiccups with the implementation, but I first wanted to be on the same page from a product perspective.

It is still pretty rough but I would love to get your thoughts 🙃@ljharb, @azz 

- I added single entrypoint for the watch plugin and then a list of options that could be "modified". I hope that this would allow the plugin to scale better. The UI is similar to https://github.com/rogeliog/jest-watch-select-projects.
- I was wondering what would be other interesting options, that we would like to add to this?
    - `--fix` is implemented and working
    - `--quiet` is still mocked, and does not do anything
- Given that they both live in the same repo, we can share state way easier, which is how the plugin and runner are communicating.
- In order to achieve this there are some small changes that are needed in `create-jest-runner`(to pass extra options to the runner) and there are no changes required in`facebook/jest`.

![watch-eslint](https://user-images.githubusercontent.com/574806/40853446-efac1482-6582-11e8-929f-0f2af23610bc.gif)

@SimenB any thoughts on this?

**Comment by SimenB** [comment]
This is insanely awesome! I think it'd be cool to attach it to `-u`, but maybe not?

>there are no changes required in`facebook/jest`

😍

**Comment by azz** [comment]
Nice! I can't think of any other ESLint flags that I would want to change during a run.

**Comment by ljharb** [comment]
What about `ext`?

**Comment by ljharb** [comment]
I’d also see the value in being able to enable/disable/reconfigure rules during the run.

**Comment by azz** [comment]
> What about ext?

As in switching from all to just `.js`? What's the use case for doing that?

> I’d also see the value in being able to enable/disable/reconfigure rules during the run.

That could be tricky if you want to persist back to disk. Would need to locate config file and patch it. Easy for JSON, YAML and standard `module.exports = {}` config files, but the JS files can be dynamic.

**Comment by ljharb** [comment]
Or from .js to `.js,.jsx` etc.

Why would you need to patch the config file? eslint on the CLI supports `--rule`

**Comment by azz** [comment]
Oh I see what you mean - ephemeral overrides, like if you want to ignore `no-console`. I'd use that.

**Comment by rogeliog** [comment]
Here is a quick prototype with "ephemeral overrides", thoughts?

NOTE: My prototype is just a node script, which is why it says "Back to Jest and running tests..."

![aaaaa](https://user-images.githubusercontent.com/574806/41116313-cd4a7314-6a3e-11e8-94ec-01e3d6e479fc.gif)

**Comment by ljharb** [comment]
That looks great for rules! what about other things like parserOptions, env, globals, "settings", things like that?

**Comment by rogeliog** [comment]
I think we could extend it as much as we want, I was thinking of choosing 2 or 3 things and polish them for the first release of the watcher(and add more later). I don't have a strong opinion about what these 2 or 3 things should be. cc: Any thoughts @ljharb @azz @SimenB?

**Comment by SimenB** [comment]
I think messing with rules makes the most sense as a first iteration - parser amd plugin stuff is more of a permanent change. Fine to revisit, of course, but a first iteration doesn't need it
